src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs

```diff
@@ -173,6 +186,27 @@ private MSBuildEvaluationResult EvaluateProjectInternal(IEngineEnvironmentSettin
                 _logger?.LogDebug("Unexpected error: {0}", e);
                 return MSBuildEvaluationResult.CreateFailure(projectPath, e.Message);
             }
+            finally
+            {
+                watch.Stop();
+                if (innerBuildWatch.IsRunning)
+                {
+                    innerBuildWatch.Stop();
+                }
+                string? targetFrameworksString = targetFrameworks != null ? string.Join(",", targetFrameworks) : targetFramework;
```

Comment:
Can you hash each of these TFMs? the msbuild `targetframeworkeval` event hashes the TFM in it, and we should be consistent.